### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false negatives for custom properties defined in reference files

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#9291]
### What minimal example or steps are needed to reproduce the bug?

`reference file`:

```css
@property --foo {
	syntax: "<color>";
	inherits: false;
	initial-value: #c0ffee;
}
```

`linted file`:

```css
a { --foo: 10px; }
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  referenceFiles: {
    files: "foo.css",
  },
  rules: {
    "declaration-property-value-no-unknown": true
  }
};
```

### How did you run Stylelint?

```shell
stylelint bar.css
```

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

I expected an error as `--foo` is typed to `<color>`

### What actually happened?

No error as `declaration-property-value-no-unknown` doesn't support `referenceFiles`

### Do you have a proposal to fix the bug?

Add support for `referenceFiles` in `declaration-property-value-no-unknown`

**Comment by jeddy3** [comment]
SGTM.

(Unsure if it's a false negative bug or a new feature - both are minor semver. Let's go with bug for now.)
